src/main/java/net/snowflake/client/core/arrow/BigIntToFixedConverter.java

```diff
@@ -137,9 +137,11 @@ public BigDecimal toBigDecimal(int index) {
   public Object toObject(int index) throws SFException {
     if (bigIntVector.isNull(index)) {
       return null;
-    } else {
-      return getLong(index);
+    } else if (!context.getSession().isJdbcArrowTreatDecimalAsInt()
```

Comment:
let's create a function `shouldTreatDecimalAsInt(context)` in `AbstractArrowVectorConverter` to have only one place where this predicate is defined (and use it in all the places where you use it currently)